### PR TITLE
Added text support

### PR DESCRIPTION
**Pull request by AlekseySh**
Added:
* `TextBaseDataset`, `TextLabeledDataset`, `TextQueryGalleryLabeledDataset`, `TextQueryGalleryDataset`, `get_mock_texts_dataset`, `visualise_text`
* Added `HFWrapper` to wrap models from HuggingFace library
* `download_mock_dataset` -> `download_mock_dataset` (the original name is also kept for back compatibility)

### CODE REVIEW

**Comment by AlekseySh** [comment]
I will add examples (code snippets) in the next PR.

File: oml/datasets/texts.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+from oml.interfaces.datasets import (
+    IBaseDataset,
+    ILabeledDataset,
+    IQueryGalleryDataset,
+    IQueryGalleryLabeledDataset,
+    IVisualizableDataset,
+)
+from oml.utils.misc import visualise_text
+
+THuggingFaceTokenizer = Any
```
Comment (leoromanovich) [inline oml/datasets/texts.py:30]: May be we should not specify tokeriner as HF? Or use specific class from tokenizers (it already installed with hf). from tokenizers import Tokenizer

Comment (leoromanovich) [reply oml/datasets/texts.py:30]: Found that comment below. I see. May be add it here too?

`# We don't use actual types programmatically in order not to bring optional dependency here`

File: oml/datasets/texts.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        self._max_length = max_length
+
+        self.input_tensors_key = input_tensors_key
+        self.index_key = index_key
+
+    def __len__(self) -> int:
+        return len(self._texts)
+
+    def __getitem__(self, item: int) -> Dict[str, Any]:
+        encoded_text = self._tokenizer(
```
[the remark this answers is not in the document]
Comment (DaloroAT) [reply oml/datasets/texts.py:67]: To get help and highlighting from the IDE if the optional library is installed, you can use this snippet (`BoolTensor` as example):

```python
from typing import TYPE_CHECKING, Any


if TYPE_CHECKING:
    try:
        from torch import BoolTensor
    except ImportError:
        BoolTensor = Any
else:
    BoolTensor = Any
```

Or raise a new error in `except ...` block with recommendations to install transformers

it should work with IDE and mypy.

File: oml/utils/misc.py [hunk 3]
```diff
@@ -162,6 +163,40 @@ def pad_array_right(arr: np.ndarray, required_len: int, val: Union[float, int])
...
+    margin = 10
+    max_width = im_size - 2 * margin
+    text_lines = []
+
+    # Split the text into lines that fit within the max_width
+    for line in text.split("\n"):
+        words = line.split()
+        current_line = []  # type: ignore
+        while words:
+            while words and draw.textlength(" ".join(current_line + [words[0]]), font=font) <= max_width:
```
Comment (DaloroAT) [inline oml/utils/misc.py:183]: you enter an infinite loop if you can't fit your word into a line of pixels with the provided font

```python
plt.imshow(visualise_text("a" * 60, (150, 150, 150)))
plt.show()
```

I don't know how realistic such a situation is, but missing 1-2 spaces and getting a long single word is not impossible, especially for raw texts

Comment (DaloroAT) [reply oml/utils/misc.py:183]: I just wanted to check how it looks and it freezes with long word

File: oml/const.py [hunk 4]
```diff
@@ -71,6 +71,7 @@ def get_cache_folder() -> Path:
 CROP_KEY = "crop"  # the format is [x1, y1, x2, y2]
 
 # Required dataset format:
```
[the remark this answers is not in the document]
Comment (AlekseySh) [reply oml/const.py:73]: You are right, done

File: oml/utils/misc.py (hunk 3 above)
Comment (AlekseySh) [reply oml/utils/misc.py:183]: you are right

I tried to cover all corner cases in the current function but gave up.

Finally, I reworked it using matplotlib and added a few tests on huge inputs.

File: oml/datasets/texts.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AlekseySh) [reply oml/datasets/texts.py:67]: okay, updated types here and for HFWrapper

File: oml/datasets/texts.py (hunk 1 above)
Comment (AlekseySh) [reply oml/datasets/texts.py:30]: okay, updated types here and for HFWrapper

File: oml/datasets/texts.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AlekseySh) [reply oml/datasets/texts.py:80]: I've spent plenty of time trying to make work this method but finally gave up. (Type hints did not work)

I've added runtime type checking instead. Here, and in `HFWrapper`

File: oml/datasets/texts.py (hunk 1 above)
Comment (AlekseySh) [reply oml/datasets/texts.py]: I've added runtime type checking
